apps/explorer/src/pages/object-result/ObjectResult.tsx

```diff
@@ -40,103 +25,46 @@ function Fail({ objID }: { objID: string | undefined }) {
-    const [showObjectState, setObjectState] = useState(DATATYPE_DEFAULT);
-    const rpc = useRpcClient();
-
-    useEffect(() => {
-        rpc.getObject({
-            id: objID,
-            options: {
-                showType: true,
-                showContent: true,
-                showOwner: true,
-                showPreviousTransaction: true,
-                showStorageRebate: true,
-            },
-        })
-            .then((objState) => {
-                const resp: DataType = translate(objState) as DataType;
-
-                const GENESIS_TX_DIGEST =
-                    'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=';
-
-                if (
-                    resp.data.tx_digest &&
-                    resp.data.tx_digest === GENESIS_TX_DIGEST
-                ) {
-                    return {
-                        ...resp,
-                        publisherAddress: 'Genesis',
-                    };
-                }
-
-                if (resp.objType === 'Move Package' && resp.data.tx_digest) {
-                    return rpc
-                        .getTransaction({
-                            digest: resp.data.tx_digest,
-                            options: {
-                                showInput: true,
-                            },
-                        })
-                        .then((txEff) => ({
-                            ...resp,
-                            publisherAddress: getTransactionSender(txEff),
-                        }))
-                        .catch((err) => {
-                            console.log(err);
-                            return resp;
-                        });
-                }
-                return resp;
-            })
-            .then((objState) => {
-                setObjectState({
-                    ...(objState as DataType),
-                    loadState: 'loaded',
-                });
-            })
-            .catch((error) => {
-                console.log(error);
-                setObjectState({ ...DATATYPE_DEFAULT, loadState: 'fail' });
-            });
-    }, [objID, rpc]);
+export function ObjectResult() {
+    const { id: objID } = useParams();
+    const { data, isLoading, isError } = useGetObject(objID!);
 
-    if (showObjectState.loadState === 'loaded') {
-        return <ObjectView data={showObjectState as DataType} />;
-    }
-    if (showObjectState.loadState === 'pending') {
+    if (isLoading) {
         return <LoadingSpinner text="Loading data" />;
     }
-    if (showObjectState.loadState === 'fail') {
+
+    if (isError) {
         return <Fail objID={objID} />;
     }
 
-    return <div>Something went wrong</div>;
-}
+    // TODO: Handle status better NotExists, Deleted, Other
+    if (data?.status !== 'Exists') {
+        return <Fail objID={objID} />;
+    }
 
-function ObjectResult() {
-    const { id: objID } = useParams();
-    const { state } = useLocation();
+    const resp = translate(data);
```

Comment:
Want to move away from this eventually but can do as part of the more redesign work.